pennylane_lightning/lightning_tensor/_tensornet.py

```
@@ -34,53 +34,132 @@
+
+    Args:
+        Mat (array): input matrix
+        site_shape (list): shape of the site tensor
+        max_bond_dim (int): maximum bond dimension
+        direction (str): direction of the SVD decomposition. Options: ["right", "left"]
+
+    Returns:
+        [array, array]: U and Vd matrices
+    """
     # TODO: Check if cutensornet allows us to remove all zero (or < tol) singular values and the respective rows and columns of U and Vd
+
     U, S, Vd = np.linalg.svd(Mat, full_matrices=False)
-    U = U * S  # Append singular values to U
+
+    # Removing noise from singular values
+    S[S < 1e-10] = 0.0
+
     bonds = len(S)
+    chi = min(bonds, max_bond_dim)
 
-    Vd = Vd.reshape([bonds] + site_shape + [-1])
-    U = U.reshape([-1] + site_shape + [bonds])
+    if direction == "right":
+        U = U * S  # Append singular values to U
+
+    if direction == "left":
+        Vd = (S * Vd.T).T  # Append singular values to Vd, equivalent operation to np.diag(S) @ Vd
+
+    # keep only chi bonds and reshape to fit the bond dimension and site shape
+    Vd = Vd[:chi].reshape([chi] + site_shape + [-1])
+    U = U[:, :chi].reshape([-1] + site_shape + [chi])
 
-    # keep only chi bonds
-    chi = min([bonds, max_bond_dim])
-    U, Vd = U[..., :chi], Vd[:chi]
     return U, Vd
 
 
-def decompose_dense(psi, n_wires, site_shape, max_bond_dim):
-    """Decompose a dense state vector/gate matrix into MPS/MPO sites."""
+def decompose_dense(
+    psi: np.ndarray,
+    n_wires: int,
+    site_shape: list[int],
+    max_bond_dim: int,
+    direction: str = "right",
+) -> list[np.ndarray]:
+    """Decompose a dense state vector/gate matrix into MPS/MPO sites.
+
+    Args:
+        psi (array): input state vector or gate matrix
+        n_wires (int): number of wires
+        site_shape (list): shape of the site tensor
+        max_bond_dim (int): maximum bond dimension
+        direction (str): direction of the SVD decomposition. Options: ["right", "left"]
```

Default to right. (why?)

---

should this be direction of canonical form?

---

should we also throw error for invalid direction?

---

Same thing here. It might be better to operate with a boolean.

---

> Default to right. (why?)

The previous version always uses the right-canonical but with this expansion now it is possible to choose. This does not have a huge impact on the internal code but after accepting external MPS becomes relevant. 

> should this be direction of canonical form?

Yes, I will replace the name to avoid confusion. 

> should we also throw error for invalid direction?

I don't think so. always should have a direction. I will use the @AmintorDusko recommendation of `use_right`

---

Done!